### ISSUE TITLE
Bug: [no-deprecated] Rule doesn't report on call inside default export

### ISSUE THREAD

**Issue by wszydlak** [typescript-eslint/typescript-eslint#10327]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.6.2&fileType=.tsx&code=PQKhAIAEBMFMAcBOsDGBDALra5oHtwA7PDcAVwGdZwMALASwvADMzCUN69DwRgAoVu07dwADwAUASnABvfuHDIMZRD1kBfANz8N-fsGDh0ZAOa0M-SVJ0GjxUifOXYY%2BHkSk4zNGQA2pNZaQA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1oBNFjpEZAIb5E3dFETRoHaJHBgAviEVA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```TypeScript
/** @deprecated do not use this function */
function x() {
  return {};
}

// caught
x();

// not caught
export default x();
```

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  rules: {
    "@typescript-eslint/no-deprecated": "error"
  },
};
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    "strictNullChecks": true
  }
}
```

### Expected Result

I expected that line `export default x();` will show error about using deprecated function, this worked properly with `eslint-plugin-deprecation`.

### Actual Result

Line `export default x();` did not show linting error.

### Additional Info

Additional demo: https://stackblitz.com/edit/tvzmwg-17xps4?file=src%2Findex.ts

**Comment by wszydlak** [comment]
Probably allowing checking nodes that are inside default exports will make it work correctly:

https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/src/rules/no-deprecated.ts#L130
